### DISCUSSION TITLE
Pedestrian wandering while waiting

### DISCUSSION THREAD

**Discussion by JulienCros** [PedestrianDynamics/jupedsim#1100] [Q&A]
Hello,

I'm currently trying to develop a tool to simulate how users would play a VR experience to optimize the path of the experience.

I was taking a look into your tools (which are great by the way) and wondering if it was possible to make the pedestrian wander randomly in the waiting area while waiting ?

Thank you

**Comment by Ozaq** [comment]
Hi @JulienCros thanks for taking interest in our project.

Unfortunately in the current release 0.9.x there is only the possibility to let agents move to a random spot inside a waiting area where they then pass the time. 

However we are currently in the process of revamping a lot of internals. For one thing we do no longer will provide an executable to run simulations, instead we will provide a C-API and a python api to do so. Part of the python API is the possibility to change the desired position of each agent after every from of the simulation. This should allow you to implement any type of waiting you desire on top of any other steering we provide. This is NOT YET available but will be soon(TM).

You can have a look at https://github.com/JuPedSim/jpscore/blob/SUMO-Integration/librarytest/test_py_jupedsim.py to get a feeling for what we want to arrive at.

All of this will be part of 0.10 a which I hope to have ready this summer. If you are interested I can update this Issue once a "steer to point (x,y)" is available in the new API. This should allow you to provide your own random walk like waiting on top of the Microscopic Agent-Agent interaction jupedsim simulates.

**Reply by pzambare3** [reply]
Hi
I am trying to integrate SUMO and JuPedSim .For that I followed the following link, :https://sumo.dlr.de/docs/Installing/Linux_Build.html#how_to_build_jupedsim_and_then_build_sumo_with_jupedsim
but I got the error with cloning. can you tell me where and how to access sumo and JuPedSim integration module.

Thank You.

**Reply by Ozaq** [reply]
Lets move this to its own discussion: https://github.com/PedestrianDynamics/jupedsim/discussions/1302

**Comment by JulienCros** [comment]
I won't really have the time to wait until summer anyway. Don't mind me, I'll keep in mind about what you said and if I can't find another solution I'll wait. Thank you for your time.
